src/actions/box.js

```diff
@@ -0,0 +1,41 @@
+import firebase, { firestore } from "../firebase";
+import { handleError } from "../utils";
+
+const BOX_ADD_ = TYPE => `BOX_ADD_${TYPE}`;
+export const BOX_ADD_START = BOX_ADD_`START`;
+export const BOX_ADD_SUCCESS = BOX_ADD_`SUCCESS`;
+export const BOX_ADD_ERROR = BOX_ADD_`ERROR`;
+
+export const addBox = ({
+  product,
+  profile,
+  organization,
+  quantity,
+  comment
+}) => dispatch => {
+  dispatch({ type: BOX_ADD_START });
+  const box = {
+    quantity,
+    comment,
+    organization: firestore.doc(organization.ref),
+    product: firestore.doc("products/" + product),
```

Comment:
A tip, which I didn't realise until after I wrote stuff this way: you can write this like `collection('products').doc(product)`, which is a bit neater.

It also means we could write `firestore.collection('organizations').doc(organization.id)` removing the need for that silly `ref` property I added.